src/main/flight/mixer.c

```diff
@@ -721,12 +722,37 @@ static void applyFlipOverAfterCrashModeToMotors(void)
     }
 }
 
+static float applyThrustLinearization(float motorOutput)
+{
+#ifdef USE_THRUST_LINEARIZATION
+    if (motorConfig()->thrustLinearization) {
+        float tl = motorConfig()->thrustLinearization * 0.01f;
+        static float a, a_reci, b, b_sq;
+        if (a != tl) {
+            a = tl;
+            a_reci = 1 / a;
+            b = (1 - a) / (2 * a);
+            b_sq = b * b;
+        }
+
+        if (motorOutput > 0.0f) {
+            motorOutput = 1.0f / fast_rsqrt( motorOutput * a_reci + b_sq ) - b;
+        }
+    }
+#endif
+    return motorOutput;
+}
+
 static void applyMixToMotors(float motorMix[MAX_SUPPORTED_MOTORS], motorMixer_t *activeMixer)
 {
     // Now add in the desired throttle, but keep in a range that doesn't clip adjusted
     // roll/pitch/yaw. This could move throttle down, but also up for those low throttle flips.
     for (int i = 0; i < motorCount; i++) {
-        float motorOutput = motorOutputMin + (motorOutputRange * (motorOutputMixSign * motorMix[i] + throttle * activeMixer[i].throttle));
+        float motorOutput = applyThrustLinearization(
```

Comment:
To me is strange always call a method `applyThrustLinearization` and inside the method an `#ifdef` decides if it is applied or not. Maybe is better give a more general name to this method or only call it when the define is active.

Reply:
That'll get optimized out. It's just to have less #ifdef's clutter the code.